packages/block-library/src/navigation-menu/index.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * `core/navigation-menu` walker.
+ * I WILL MOVE THIS FROM HERE!
+ */
+class Walker_Simple_Example extends Walker {
+
+	/**
+	 * Set the properties of the element which give the ID of the current item and its parent.
+	 *
+	 * @var $db_fields array
+	 */
+	var $db_fields = array(
+		'parent' => 'parent',
+		'id' => 'id'
+	);
+
+	function start_lvl(&$output, $depth=0, $args=array()) {
+		$output .= "\n<ul>\n";
+	}
+
+
+	function end_lvl(&$output, $depth=0, $args=array()) {
+		$output .= "</ul>\n";
+	}
+
+
+	function start_el(&$output, $item, $depth=0, $args=array(), $current_object_id = 0) {
+		$output .= "<li>".$item->title[0];
+    }
+
+
+	function end_el(&$output, $item, $depth=0, $args=array()) {
+		$output .= "</li>";
+	}
+}
+
+/**
+ * Server-side rendering of the `core/navigation-menu` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/navigation-menu` block on server.
+ *
+ * @param array $attributes The block attributes.
+ * @param array $content The saved content.
+ * @param array $block The parsed block.
+ *
+ * @return string Returns the post content with the legacy widget added.
+ */
+function render_block_navigation_menu( $attributes, $content, $block ) {
+	$block  = prepare_navigation_for_walker( $block );
```

Comment:
What's the benefit of using a Walker? I'm wondering if we should just use our own recursive function which works with nested arrays. That lets us remove this preparation step.

Reply:
Yea, I resolved to not using a walker and instead opting for default markup and a recursive list generator. We'll see how this changes depending on where the menu block is headed about themes.